### DISCUSSION TITLE
VAMPNet results

### DISCUSSION THREAD

**Discussion by kuntalg97** [deeptime-ml/deeptime#314] [Q&A] [answered]
Hi,

I am analyzing a large system by examining the tilt angles between different protein oligomers roughly along a plane. I have performed TICA analysis using these tilt angles as features, and I can see distinct TICA states.

I now wish to extend this analysis using VAMPNets. However, I am facing an issue with reproducibility—VAMPNet states vary significantly when I use different random seeds. I am concerned about the interpretation of these results due to this variability.

Is there a way to fix this?

Thanks!

**Comment by clonker** [comment, marked as answer]
Hey, thank you for the question!

Some seed-dependence is normal for neural network methods (due to SGD and its variants), and it does not necessarily invalidate the results. You should look at the vamp score of the learned transformations and compare them first. Training probably finds different local optima. It is of course only a proxy and there may be many different ways to project something onto slow processes even under a similar score. But it might give you an idea of the quality of learned results. Make sure to evaluate these scores on a validation set. You can also compare them to the VAMP score of your TICA transformation.
Depending on your network setup it might also be that you need more regularization and/or different training hyperparameters.
Generally I'd rather recommend to continue the analysis from the TICA/VAMPNet projection and build a MSM + PCCA to obtain a state discretization. For VAMPNets you can also use softmax as output nonlinearity, that already gives you something like a fuzzy PCCA decomposition into states (as done [here](https://deeptime-ml.github.io/latest/notebooks/examples/ala2-example.html), although, of course, this is alanine dipeptide and everything works well with that system. More complex systems usually require more data and/or more tuning.

**Comment by thempel** [comment]
Agreed with @clonker. You might want to check what the slow coordinates picked up by TICA or VAMPNets mean structurally, and how consistent they are among different runs. There might be some iterations necessary between feature definition and model estimation. 

One more side-remark: Be aware that angles, if used directly, can cause trouble in this analysis because they are periodic. TICA or VAMPnets might pick up rare jumps between, say, `-pi` and `pi` that are really very small. Maybe consider using sine and/or cosine of these angles.
